src/test/analyzer_test.ts

```diff
@@ -523,4 +546,211 @@ suite('Analyzer', () => {
           testSeed.events.map(e => e.name), ['fired-event', 'data-changed']);
     });
   });
+
+  suite.skip('race conditions and caching', () => {
+
+    class RacyUrlLoader implements UrlLoader {
+      constructor(
+          public map: Map<string, string>,
+          private waitFunction: () => Promise<void>) {
+      }
+      canLoad() {
+        return true;
+      }
+      async load(path: string) {
+        await this.waitFunction();
+        const result = this.map.get(path);
+        if (result != null) {
+          return result;
+        }
+        throw new Error(`no known contents for ${path}`);
+      }
+    }
+
+    const editorSimulator = async(waitFn: () => Promise<void>) => {
+      // Here we're simulating a lot of noop-changes to base.html, which has
+      // two imports, which mutually import a common dep. This stresses the
+      // analyzer's caching.
+
+      const contentsMap = new Map<string, string>([
+        [
+          'base.html',
+          `<link rel="import" href="a.html">\n<link rel="import" href="b.html">`
+        ],
+        ['a.html', `<link rel="import" href="common.html">`],
+        ['b.html', `<link rel="import" href="common.html">`],
+        ['common.html', `<custom-el></custom-el>`],
+      ]);
+      const analyzer =
+          new Analyzer({urlLoader: new RacyUrlLoader(contentsMap, waitFn)});
+      const promises: Promise<Document>[] = [];
+      for (let i = 0; i < 30; i++) {
+        await waitFn();
+        for (const key of contentsMap) {
+          if (Math.random() > 0.5) {
+            analyzer.analyze(key[0], key[1]);
+          }
+        }
+        promises.push(analyzer.analyze('base.html'));
+      }
+      const documents = await Promise.all(promises);
+      for (const document of documents) {
+        const imports = Array.from(document.getByKind('import'));
+        assert.deepEqual(
```

Comment:
After the `map()` this looks like `assert.sameMembers()`. Might give a better error message.